doc/source/contributing.rst

```diff
@@ -103,167 +107,164 @@ want to clone your fork to your machine::
 
     git clone https://github.com/your-user-name/pandas.git pandas-yourname
     cd pandas-yourname
-    git remote add upstream git://github.com/pandas-dev/pandas.git
+    git remote add upstream https://github.com/pandas-dev/pandas.git
 
 This creates the directory `pandas-yourname` and connects your repository to
 the upstream (main project) *pandas* repository.
 
-Creating a branch
------------------
+.. _contributing.dev_env:
 
-You want your master branch to reflect only production-ready code, so create a
-feature branch for making your changes. For example::
+Creating a development environment
+----------------------------------
 
-    git branch shiny-new-feature
-    git checkout shiny-new-feature
+To test out code changes, you'll need to build pandas from source, which
+requires a C compiler and python environment. If you're making documentation
+changes, you can skip to :ref:`contributing.documentation` but you won't be able
+to build the documentation locally before pushing your changes.
 
-The above can be simplified to::
+.. _contributiong.dev_c:
 
-    git checkout -b shiny-new-feature
+Installing a C Complier
+~~~~~~~~~~~~~~~~~~~~~~~
 
-This changes your working directory to the shiny-new-feature branch.  Keep any
-changes in this branch specific to one bug or feature so it is clear
-what the branch brings to *pandas*. You can have many shiny-new-features
-and switch in between them using the git checkout command.
+Pandas uses C extensions (mostly written using Cython) to speed up certain
+operations. To install pandas from source, you need to compile these C
+extensions, which means you need a C complier. This process depends on which
+platform you're using. Follow the `CPython contributing guidelines
+<https://docs.python.org/devguide/setup.html#build-dependencies>`_ for getting a
+complier installed. You don't need to do any of the ``./configure`` or ``make``
+steps; you only need to install the complier.
 
-To update this branch, you need to retrieve the changes from the master branch::
+For Windows developers, the following links may be helpful.
 
-    git fetch upstream
-    git rebase upstream/master
+- https://blogs.msdn.microsoft.com/pythonengineering/2016/04/11/unable-to-find-vcvarsall-bat/
+- https://github.com/conda/conda-recipes/wiki/Building-from-Source-on-Windows-32-bit-and-64-bit
+- https://cowboyprogrammer.org/building-python-wheels-for-windows/
```

Comment:
we should prob update these and eliminate the older ones

Reply:
Yes, I'll go through these later today and remove the old / outdated ones.